### DISCUSSION TITLE
GPU info not showing

### DISCUSSION THREAD

**Discussion by KerimSaman** [BartoszCichecki/LenovoLegionToolkit#1488] [Q&A]
![Ekran görüntüsü 2024-11-13 202128](https://github.com/user-attachments/assets/f55d0c2a-a770-42e5-b56a-4075ecb38839)
Sometimes my GPU info doesn't show up, it's not disabled and I'm using it with the monitor. I don't remember if it's connected to the last update. Is this a problem or is it me?

**Comment by Ace-Radom** [comment]
*Sometimes* doesn't help tbh. If this issue occurs again plz catch a log and upload it here, and then we can find out what's going wrong.

**Reply by KerimSaman** [reply]
> _Sometimes_ doesn't help tbh. If this issue occurs again plz catch a log and upload it here, and then we can find out what's going wrong.

How do I get a log?

**Reply by Ace-Radom** [reply]
See section [How to collect logs](https://github.com/BartoszCichecki/LenovoLegionToolkit?tab=readme-ov-file#how-to-collect-logs) in README, there's a pretty nice and clear guide. Remember closing the running LLT completely before catching logs is the only thing you need to make sure.

You can upload your log in this thread and I (or maybe bc) will take a look later. If this is truly a bug in LLT we'll convert this discussion into an issue.

**Reply by KerimSaman** [reply]
It works fine when trying to get logs, interesting. I will share if I manage to get a log when it is not working.

**Reply by KerimSaman** [reply]
I discovered something interesting, if I remove the Discrete GPU section that I checked in the customize area, the GPU information above disappears (after restart). 
![Ekran görüntüsü 2024-11-18 140824](https://github.com/user-attachments/assets/8eb5c8c3-d5de-4e5f-a1c9-5ecfb26eff69)
![Ekran görüntüsü 2024-11-18 141844](https://github.com/user-attachments/assets/491700a4-8482-47cf-ac29-004e216722a3)

**Reply by Ace-Radom** [reply]
As I said, if you can reproduce this issue, plz catch a log and upload it. The thing you found out may be useful for trouble shooting, but just with two screenshots I cannot reproduce it on my device.

I only tried to deactivate the dGPU on my laptop under hybrid mode (just noticed you're in hybrid mode, I'm not familiar with Turkish) and I got sth similar to your situation.

![image](https://github.com/user-attachments/assets/ac796dea-ad1e-4c2f-a33f-5f93cddce5ec)

dGPU is powered off may be the reason, but anyway, plz attach log.

Plus: you can just redo the things you did before (like the two screenshots above), but with logging. You don't need to worry about it's not the same thing you mentioned in your original question, cuz how to analyse the log is my thing. If the log doesn't figure out what's happening I would tell you what to do next, but I think if you do the things above again, with log, it should be enough.

**Reply by KerimSaman** [reply]
Yes, it is normal that the information does not appear if the GPU turns off in hybrid mode, it was like that before. But now I use a monitor and the GPU is always on. I had turned off the Discrate GPU section from customize and there was no GPU information. I closed the application.
1- I opened the app in log capture format, the GPU info part was not working.
2- I added the Discrate GPU section from Customize.
3- After it was added, the information started to come. 
I closed the program and I am sharing the log file with you. 

[log_2024_11_18_13_55_54.txt](https://github.com/user-attachments/files/17801341/log_2024_11_18_13_55_54.txt)

**Reply by Ace-Radom** [reply]
That's a bit weird. From the log you caught I saw no gpu datas were got in your step one, but there was also no conspicuous error there. I do have a thought what's happening but I don't have time to verify it today or tomorrow (likely). I'll come back to you when I finish my personal stuffs.

Short info update: things take more time than I thought. I didn't forget this and I'll try to make myself free for this one in the coming weekend.

**Reply by KerimSaman** [reply]
That's not a problem for me, you can deal with it anytime you want. Thank you

**Reply by Ace-Radom** [reply]
![image](https://github.com/user-attachments/assets/781b4747-9d8d-4634-9689-d0d34ffe7bde)

Status update: I successfully reproduced this bug on my device.
This is because `GPUController` hasn't got initialized on startup when you remove `Discrete GPU` from dashboard. I need sometime to go through all GPU & sensor control codes since I've never worked on it before, but it is a bug in LLT. I'll make an issue related to this later.

**Reply by KerimSaman** [reply]
I am happy to discover a real problem 😊. I wish you good work on this and thank you for your concern.

**Reply by Ace-Radom** [reply]
> I am happy to discover a real problem 😊. I wish you good work on this and thank you for your concern.

No problem, also thanks for the catch. I may post a test build installer under #1501 later for you to test. If you have tried it plz report the result under the issue to let me see if it is also solved on your device.

**Reply by KerimSaman** [reply]
> > I am happy to discover a real problem 😊. I wish you good work on this and thank you for your concern.
> 
> No problem, also thanks for the catch. I may post a test build installer under #1501 later for you to test. If you have tried it plz report the result under the issue to let me see if it is also solved on your device.

Definitely.

**Comment by BartoszCichecki** [comment]
Closing as PR was merged.
